pkg/machines/components/create-vm-dialog/createVmDialog.jsx

```diff
@@ -279,66 +280,75 @@ const SourceRow = ({ source, sourceType, networks, nodeDevices, providerName, on
     );
 };
 
-const OSRow = ({ vendor, osInfoList, os, vendors, onValueChanged, validationFailed }) => {
-    const validationStateOsVendor = validationFailed.vendor ? 'error' : undefined;
-    const familyList = vendors.familyList;
-    const familyMap = vendors.familyMap;
-    const vendorMap = vendors.vendorMap;
-    const vendorSelectEntries = [];
-
-    if (familyMap[DIVIDER_FAMILY]) {
-        vendorSelectEntries.push((
-            <Select.SelectEntry data={NOT_SPECIFIED} key={NOT_SPECIFIED}>{_(NOT_SPECIFIED)}</Select.SelectEntry>));
-        vendorSelectEntries.push((<Select.SelectDivider key='divider' />));
-    }
+class OSRow extends React.Component {
+    constructor(props) {
+        super(props);
+        const IGNORE_VENDORS = ['ALTLinux', 'Mandriva', 'GNOME Project'];
+        const osInfoListExt = this.props.osInfoList
+                .filter(os => !IGNORE_VENDORS.find(vendor => vendor == os.vendor))
+                .map(os => correctSpecialCases(os))
+                .filter(os => filterReleaseEolDates(os))
```

Comment:
`filter->map->filter` could this be done with `map->filter` or just one `reduce`?

Reply:
Done.